File: dataStructures&algorithms/algorithm/leetCode/aboutLinkedList/deleteNode/deleteNode.php
<?php
/**
 * @date: 2022/3/13
 * @createTime: 7:38 PM
 */

// Definition for singly-linked list.
class ListNode
{
    public $val = 0;
    public $next = null;

    function __construct($val)
    {
        $this->val = $val;
    }
}

// Leetcode-237
class Solution
{
    function deleteNode(ListNode $node)
    {
        if (!isset($node)) {
            return;
        }
        if (!isset($node->next)) {
            unset($node);
            return;
        }
        $node->val = $node->next->val;
        $delNode = $node->next;
        $node->next = $delNode->next;
        unset($delNode);
    }
}

File: dataStructures&algorithms/algorithm/leetCode/aboutLinkedList/deleteNode/deleteNode.go
/**
 * @Author: shershon
 * @Description:
 * @Date: 2022/03/13 7:46 PM
 */

package main

// DelNode Definition for singly-linked list.
type DelNode struct {
	Val  int
	Next *DelNode
}

// Leetcode-237
func deleteNode(node *DelNode) {
	if node == nil {
		return
	}
	if node.Next == nil {
		node = nil
		return
	}
	node.Val = node.Next.Val
	delNode := node.Next
	node.Next = delNode.Next

	return
}
